server/neptune/workflows/internal/deploy/workflow.go

```diff
@@ -98,28 +98,37 @@ func newRunner(ctx workflow.Context, request Request, tfWorkflow terraform.Workf
 	}
 	revisionQueue := queue.NewQueue(func(ctx workflow.Context, d *queue.Deploy) {
 		lockStateUpdater.UpdateQueuedRevisions(ctx, d)
-	}, metricsHandler)
+	}, scope)
 
 	worker, err := queue.NewWorker(ctx, revisionQueue, scope, a, tfWorkflow, request.Repo.FullName, request.Root.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	revisionReceiver := revision.NewReceiver(ctx, revisionQueue, a, sideeffect.GenerateUUID, worker)
+	revisionReceiver := revision.NewReceiver(ctx, revisionQueue, a, sideeffect.GenerateUUID, worker,
+		scope.SubScopeWithTags(map[string]string{
+			workflowMetrics.SignalNameTag: NewRevisionSignalID,
+		}))
 
 	return &Runner{
 		Queue:                    revisionQueue,
 		Timeout:                  RevisionReceiveTimeout,
 		QueueWorker:              worker,
 		RevisionReceiver:         revisionReceiver,
 		NewRevisionSignalChannel: workflow.GetSignalChannel(ctx, NewRevisionSignalID),
-		MetricsHandler:           metricsHandler,
+		Scope:                    scope,
 	}, nil
 }
 
+func (r *Runner) shutdown() {
+	r.Scope.Gauge(ActiveDeployWorkflowStat).Update(0)
+	r.Scope.Counter(SuccessDeployWorkflowStat).Inc(1)
```

Comment:
Is it a guarantee that it will be successful? Are cancels successful?

Reply:
I guess that's a good point, but are they errors? I guess we don't know technically. I would need to think about how to detect manual terminations and track those separately.  For now, I'll just merge this to keep it consistent with the terraform workflow stats.